### DISCUSSION TITLE
Error: connect EADDRNOTAVAIL 127.0.0.1 when Prisma deployed to Azure function app with vnet integration

### DISCUSSION THREAD

**Discussion by inertica** [prisma/prisma#5835] [Q&A] [answered]
Hello,

we have a data exporter function app using Prisma 2 + MySQL database server deployed to Azure production environment with a virtual private network. Database connection has been working properly until the beginning of February. Prisma version 2.10.2 has been used before successfully, but even that version does not work anymore. This is the error message with the port number the changing in error message on different export runs:

```
2021-02-25T09:21:33.515 [Error] Executed 'Functions.offline-data-exporter' (Failed, Id=b4889147-xxxx-xxxx-xxx-
xxxxxxxxxxx, Duration=1554ms)
Result: FailureException: Error:Invalid `prisma.disconnector.findMany()` invocation 
in D:\home\site\wwwroot\dist\shared\offline-data-exporter.js:15:84
connect EADDRNOTAVAIL 127.0.0.1 - Local (0.0.0.0:50306)Stack: Error:Invalid `prisma.disconnector.findMany()`
invocation in D:\home\site\wwwroot\dist\shared\offline-data-exporter.js:15:84
connect EADDRNOTAVAIL 127.0.0.1 - Local (0.0.0.0:50306)at PrismaClientFetcher.request 
(D:\home\site\wwwroot\node_modules\@prisma\client\runtime\index.js:78585:15)at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

The database server is on 10.0.0.x and pinging with `tcpping` to MySQL database port 3306 (or http port 80) from Azure function app console works. It seems that the connection to database is established, but the query or results cannot be processed correctly. 

MySQL connection string is in format `mysql://user:pass@10.0.0.x/database?connection_limit=1&sslmode=disable`

The function app is also deployed to Azure sandbox and QA environments without vnet integration, and there are no problems. These environments use Azure MySQL for database, so they are unfortunately not comparable to production.

The following have been checked:
* database connection string is ok and database user can connect to database and has sufficient rights
* deployed function app stopped and restarted 
* database works ok using the mysql2 library
* vnet subnet 10.0.100.x is connected and working correctly on other function apps

The following have been tried and none of this have worked:
* upgrading Prisma version to 2.17.0 and re-deploying the function app
* downgrading back to 2.10.2 and re-deploying the function app
* building release from the last known working commit and re-deploying the function app
* using different queries and reducing the data export to one simple query
* deploying code to new function app with similar configuration built from ground up
* removing `connection_limit` and `sslmode` from MySQL connection string

Has anybody had similar problems and what could be the possible cause and cure for that? We're consulted some database and Azure professionals and are unable to come up with a solution. The strangest thing that the function app has been working previously without problems, and other function apps not using Prisma to connect to the same database server on the vnet work fine.

**Comment by nikolasburk** [comment, marked as answer]
Hey, thank you so much for this detailled write-up! Do you mind [reposting this as an actual issue](https://github.com/prisma/prisma/issues/new/choose)? That way our Engineering team can start to look into this and create a reproduction! 🙏

**Reply by inertica** [reply]
Hello @nikolasburk, created an issue from this: https://github.com/prisma/prisma/issues/5844

Thanks for the quick reply.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
